src/basic/process-util.c

```diff
@@ -140,14 +139,10 @@ static int get_process_cmdline_nulstr(
          * If PROCESS_CMDLINE_COMM_FALLBACK is specified in flags and the process has no command line set
          * (the case for kernel threads), or has a command line that resolves to the empty string, will
          * return the "comm" name of the process instead. This will use at most _SC_ARG_MAX bytes of input
-         * data.
-         *
-         * Returns an error, 0 if output was read but is truncated, 1 otherwise.
-         */
+         * data. */
 
         p = procfs_file_alloca(pid, "cmdline");
-        r = read_virtual_file(p, max_size, &t, &k); /* Let's assume that each input byte results in >= 1
-                                                     * columns of output. We ignore zero-width codepoints. */
+        r = read_virtual_file(p, SIZE_MAX, &t, &k);
```

Comment:
I don't think this is a good idea. This effectively disables the size limit, which means that we can read (and parse) megabytes of data here. If the data is full of short sequences, this will lead to a fairly significant "allocation amplification" when we split this in individual strings. This function is called in places that lists processes, also for untrusted processes, so I think it should be more careful.